jupyterlab_search_replace/search_engine.py

```diff
@@ -50,6 +50,8 @@ class SearchEngine:
     The implementation is using `ripgrep <https://github.com/BurntSushi/ripgrep>`_.
     """
 
+    search_task: ClassVar[Optional[asyncio.Task]] = None
```

Comment:
let's add some documentation

```suggestion
    # Keep track of the previous search task to run only one task at a time
    search_task: ClassVar[Optional[asyncio.Task]] = None
```